Controllers/Backend/ConnectConfig.php

```diff
@@ -462,24 +463,107 @@ public function getConnectExport()
     /**
      * It will make a call to SocialNetwork to reset the price type,
      * if this call return success true, then it will reset the export settings locally
+     *
+     * @return void
      */
     public function resetPriceTypeAction()
     {
-        $response = $this->getSnHttpClient()->sendRequestToConnect(array(), 'account/reset/price-type');
+        $response = $this->getSnHttpClient()->sendRequestToConnect('account/reset/price-type');
+        $responseBody = json_decode($response->getBody());
+
+        if(!$responseBody->success) {
+            $this->View()->assign([
+                'success' => false,
+                'message' => $responseBody->message
+            ]);
+            return;
+        }
+
+        try{
+            $this->resetExportedProducts();
+        } catch (\Exception $e){
+            $this->View()->assign([
+                'success' => false,
+                'message' => $responseBody->message
+            ]);
+            return;
+        }
+
+        $this->View()->assign([
+            'success' => true,
+        ]);
+    }
+
+    /**
+     * It will make a call to SocialNetwork to reset the exchange settings,
+     * if this call return success true, then it will reset the export settings locally
+     *
+     * @return void
+     */
+    public function resetExchangeSettingsAction()
+    {
+        $response = $this->getSnHttpClient()->sendRequestToConnect('account/reset/exchange-settings');
+
         $responseBody = json_decode($response->getBody());
 
         if(!$responseBody->success) {
             $this->View()->assign([
                 'success' => false,
                 'message' => $responseBody->message
             ]);
+            return;
+        }
+
+        try{
+            $this->resetExportedProducts();
+
+            //remove the existing api key
+            $this->getConfigComponent()->setGeneralConfigs(['apiKey' => '']);
 
+            //recreate the register menu
+            $this->createRegisterMenu();
+
+        } catch (\Exception $e){
+            $this->View()->assign([
+                'success' => false,
+                'message' => $responseBody->message
+            ]);
             return;
         }
 
-        // WARNING This code remove the current product changes
-        // This is a single call operation and its danger one
-        // This code should not be used anywhere
+        $this->View()->assign([
+            'success' => true,
+        ]);
+    }
+
+    /**
+     * @throws \Doctrine\DBAL\Exception\InvalidArgumentException
+     */
+    private function createRegisterMenu()
```

Comment:
Please move this in service, existing one or just create new service for Connect menus.